src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
I'm not a fan of the getLastActionStatus(action_t last_action) method for accessing the status because it seems more complicated than necessary. What do you think of the alternative idea of modifying the [act function](https://github.com/LARG/HFO/blob/master/src/HFO.hpp#L42) so that it returns a boolean - True if the action may have been successful, False if definitely not successful.